### PR TITLE
Alter how set_members works in the new solaris group provider

### PR DESCRIPTION
**Pull request by jjustice6**
Signed-off-by: Joshua Justice <jjustice6@bloomberg.net>

### Description

I had previously submitted #8165, but there were issues with the `set_members` function there.
Solaris manpage reference: https://docs.oracle.com/cd/E26502_01/html/E29031/groupmod-1m.html

### Issues Resolved

Discussion with @tas50 about excluded_members being an issue.

### Check List

- [x] New functionality includes tests
- [x] All tests pass
- [ ] RELEASE\_NOTES.md has been updated if required (not required for bugfixes, required for API changes)
- [X] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: spec/unit/provider/group/solaris_spec.rb [hunk 1]
```diff
@@ -48,7 +48,7 @@
       platforms = {
         "solaris2" => [ "-G" ],
       }
-
+
```
Comment (tas50) [inline spec/unit/provider/group/solaris_spec.rb]: this is probably going to fail chefstyle

Comment (jjustice6) [reply spec/unit/provider/group/solaris_spec.rb]: it did indeed

File: spec/functional/resource/group_spec.rb [hunk 2]
```diff
@@ -292,14 +292,16 @@ def create_group
     end
   end
 
-  let(:group_name) { "group#{SecureRandom.random_number(9999)}" }
+  # Groups below 1000 are reserved on a number of OSes, pick higher numbers
```
Comment (tas50) [inline spec/functional/resource/group_spec.rb]: Change this comment to below 100

File: spec/functional/resource/group_spec.rb [hunk 3]
```diff
@@ -292,14 +292,16 @@ def create_group
...
+  let(:number) { SecureRandom.random_number(9899) + 100 }
+  let(:group_name) { "grp#{number}" } # groupname 8 characters or less
```
Comment (tas50) [inline spec/functional/resource/group_spec.rb]: Can you mention we're doing <= 8 for Solaris

Comment (jjustice6) [reply spec/functional/resource/group_spec.rb]: done, and a reference added that it's not just solaris with the <= 8 limit

File: spec/functional/resource/group_spec.rb (hunk 2 above)
Comment (jjustice6) [reply spec/functional/resource/group_spec.rb]: done

**Comment by tas50** [comment]
Closing in favor of #8226

**Pull request closed without merging**